packages/semcom-demo/src/app/connect/connect.component.ts

```diff
@@ -0,0 +1,27 @@
+import { AppState, ConnectState } from '../app.reducers';
+import { Component, OnInit } from '@angular/core';
+import { ConnectPageInit, ProviderSelected } from './connect.actions';
+import { Observable } from 'rxjs';
+import { Provider } from './models/provider.model';
+import { Store } from '@ngrx/store';
+
+@Component({
+  selector: 'demo-connect',
+  templateUrl: './connect.component.html',
+  styleUrls: ['./connect.component.scss']
+})
+export class ConnectComponent implements OnInit{
+
+    providers$: Observable<Provider[]> = this.store.select<Provider[]>(state => state.connectState.providers);
+
+    constructor(private store: Store<AppState>) {}
+
+    ngOnInit(): void {
+      this.store.dispatch(ConnectPageInit());
+    }
+
+    connect(provider: Provider): void {
```

Comment:
Don't forget to check for `null`.

Reply:
Aha, right ... I was under the impression that Typescript makes that unneccessary. That's not the case?